src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -139,6 +155,16 @@ public sealed class DiagnosticDescriptorCreationAnalyzer : DiagnosticAnalyzer
             description: s_localizableProvideCustomTagsDescription,
             customTags: WellKnownDiagnosticTags.Telemetry);
 
+        public static readonly DiagnosticDescriptor DoNotUseReservedDiagnosticIdRule = new DiagnosticDescriptor(
+            DiagnosticIds.DoNotUseReservedDiagnosticIdRuleId,
+            s_localizableDoNotUseReservedDiagnosticIdTitle,
+            s_localizableDoNotUseReservedDiagnosticIdMessage,
+            DiagnosticCategory.MicrosoftCodeAnalysisDesign,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
```

Comment:
I think we all wanted this to be an `Error` severity by default.

Reply:
Diagnostic analyzers cannot use Error as a default severity. I filed #3159 to fix this.